sdk/forms/src/main/java/com/klaviyo/forms/BridgeMessage.kt

```diff
@@ -0,0 +1,163 @@
+        private const val IAF_TYPE_VERSION_KEY = "version"
+
+        private const val IAF_MESSAGE_HAND_SHOOK = "handShook"
+        private const val IAF_MESSAGE_TYPE_SHOW = "formWillAppear"
+        private const val IAF_MESSAGE_TYPE_AGGREGATE_EVENT = "trackAggregateEvent"
+        private const val IAF_MESSAGE_TYPE_PROFILE_EVENT = "trackProfileEvent"
+        private const val IAF_MESSAGE_TYPE_DEEPLINK = "openDeepLink"
+        private const val IAF_MESSAGE_TYPE_CLOSE = "formDisappeared"
+        private const val IAF_MESSAGE_TYPE_ABORT = "abort"
+
+        /**
+         * Handshake data: the types and version numbers of bridge messages that the SDK supports
+         */
+        internal val handShakeData by lazy {
+            JSONArray(
+                listOf(
+                    mapOf(
+                        IAF_MESSAGE_TYPE_KEY to IAF_MESSAGE_HAND_SHOOK,
+                        IAF_TYPE_VERSION_KEY to 1
+                    ),
+                    mapOf(
+                        IAF_MESSAGE_TYPE_KEY to IAF_MESSAGE_TYPE_SHOW,
+                        IAF_TYPE_VERSION_KEY to 1
+                    ),
+                    mapOf(
+                        IAF_MESSAGE_TYPE_KEY to IAF_MESSAGE_TYPE_AGGREGATE_EVENT,
+                        IAF_TYPE_VERSION_KEY to 1
+                    ),
+                    mapOf(
+                        IAF_MESSAGE_TYPE_KEY to IAF_MESSAGE_TYPE_PROFILE_EVENT,
+                        IAF_TYPE_VERSION_KEY to 1
+                    ),
+                    mapOf(
+                        IAF_MESSAGE_TYPE_KEY to IAF_MESSAGE_TYPE_DEEPLINK,
+                        IAF_TYPE_VERSION_KEY to 1
+                    ),
+                    mapOf(
+                        IAF_MESSAGE_TYPE_KEY to IAF_MESSAGE_TYPE_CLOSE,
+                        IAF_TYPE_VERSION_KEY to 1
+                    ),
+                    mapOf(
+                        IAF_MESSAGE_TYPE_KEY to IAF_MESSAGE_TYPE_ABORT,
+                        IAF_TYPE_VERSION_KEY to 1
+                    )
+                )
+            ).toString()
+        }
+
+        /**
+         * Parse a native bridge message string into a [BridgeMessage]
+         *
+         * @throws IllegalStateException for unexpected message strings
+         */
+        fun decodeWebviewMessage(message: String): BridgeMessage {
+            val jsonMessage = JSONObject(message)
+            val jsonData = jsonMessage.optJSONObject(IAF_MESSAGE_DATA_KEY) ?: JSONObject()
+
+            return when (val type = jsonMessage.optString(IAF_MESSAGE_TYPE_KEY)) {
+                IAF_MESSAGE_HAND_SHOOK -> HandShook
+
+                IAF_MESSAGE_TYPE_SHOW -> Show(
+                    formId = jsonData.optString("formId").ifEmpty { "" }
+                )
+
+                IAF_MESSAGE_TYPE_AGGREGATE_EVENT -> AggregateEventTracked(
+                    payload = jsonData
+                )
+
+                IAF_MESSAGE_TYPE_PROFILE_EVENT -> ProfileEvent(
```

Comment:
Profile events werent in the MVP right? With this code bridged, whats the current functionality whenever a profile event is triggered? Or is it that the onsiteJS bridge is where the profile-level event handling is missing so this will never fire for now?

Reply:
IAF editor only supports close and deep link, neither of which trigger a profile event yet. But by supporting it in the SDK we leave the option open for a fast follow of adding some profile events via a fender update.

Reply:
Ah perfect thanks for clarifying that.

Reply:
Only concern is once we support profile level events in the future, are we confident that the logic here would lead to some bugs popping up in this upcoming SDK release? Like the code for supporting profile level events is there now so its forward thinking, but we've no guarantee that it'll work and it may be more confusing for users if that was the case (even if we did not advertise it as a feature for this version)

Reply:
Oh no, we've already tested profile event creation e2e. I made a fender branch with a profile level event triggered by deep link, asset-sourced that, and verified that the profile event made it to the backend. 
FWIW if we did find a bug like that, we would be able to use our handshake data to selectively exclude a buggy SDK implementation for even one message bridge type!